packages/scene-composer/src/components/toolbars/floatingToolbar/items/HistoryItemGroup.tsx

```diff
@@ -6,14 +6,18 @@ import { useStore } from '../../../../store';
 import { UndoStoreState } from '../../../../store/middlewares';
 import { ToolbarItem } from '../../common/ToolbarItem';
 import { ToolbarItemGroup } from '../../common/styledComponents';
+import useDynamicScene from '../../../../hooks/useDynamicScene';
 
 export function HistoryItemGroup() {
   const sceneComposerId = useContext(sceneComposerIdContext);
   const [redo, undo] = useStore(sceneComposerId)((state) => [state.redo, state.undo]);
   const undoStore = useStore(sceneComposerId)((state) => state.undoStore);
   const [undoState, setUndoSate] = useState<UndoStoreState | undefined>(undoStore?.getState());
-  const isRedoEnabled = (undoState?.futureStates && undoState.futureStates.length > 0) || false;
-  const isUndoEnabled = (undoState?.prevStates && undoState.prevStates.length > 0) || false;
+  const dynamicSceneEnabled = useDynamicScene();
+
+  // Temporarily disable undo/redo when dynamic scene is enabled
+  const isRedoEnabled = undoState?.futureStates && undoState.futureStates.length > 0 && !dynamicSceneEnabled;
```

Comment:
Nit-pick: You could clean this up a little by implementing some optional chaining. 
`const isRedoEnabled = undoState?.futureStates?.length > 0 && !dynamicSceneEnabled;`